### PR TITLE
[chore] Add release, support, and deprecation docs

### PR DESCRIPTION
**Pull request by shaneutt**
This PR adds several bits of documentation which helps to improve utilizing and contributing to this repository.

This resolves [DCOS-62869](https://jira.mesosphere.com/browse/DCOS-62869).

### CODE REVIEW

**Comment by shaneutt** [comment]
Blocked: must have signoff from @jaxesn & @samba before merging, specifically due to the support language added here.

File: README.md [hunk 1]
```diff
@@ -13,9 +13,66 @@ You will find the following directories here:
...
+
+Creating a releases is as simple as cutting a tag and making a [Github release](https://help.github.com/en/github/administering-a-repository/creating-releases).
+
+If the release is meant to be an official KSphere supported release, ensure it's not marked as a pre-release, follows the `v{MAJOR}.{MINOR}.{PATCH}` [semver](https://semver.org/) pattern, and ensure the language in the title and description indicate "Official Release".
+
+Supported releases are "supported" via the terms of support for some other KSphere entity using them, particularly releases will often be connected with a [Konvoy Release](https://github.com/mesosphere/konvoy). Make sure that you mention and link to any externally related sources for your release in the release description (e.g. if the release is specifically intended to support a specific Konvoy release, say so in the description).
+
+For all other non-official releases, make sure your tag and description are distinctly different from the official release pattern, explain the purpose of your release, and mark is as a `pre-release`.
+
+# Releases
```
Comment (juliangieseke) [inline README.md]: duplicate headline

**Comment by juliangieseke** [comment]
Overall LGTM - would love to have some clarifications around not changing existing revisions and that its better to create new ones instead.

**Comment by makkes** [comment]
@shaneutt this seems like a good moment to add a license to this repo. wdyt? @mattj-io, want to chime in?

**Comment by shaneutt** [comment]
> @shaneutt this seems like a good opportunity to add a license to this repo. wdyt? @mattj-io, want to chime in?

@samba as the PO would need to chime in on this, but my understanding is that at least for the moment we want to be closed source but with public visibility.

cc @joejulian

File: README.md [hunk 2]
```diff
@@ -13,9 +13,50 @@ You will find the following directories here:
...
+
+While you may see other tags in our [releases page](https://github.com/mesosphere/kubernetes-base-addons/releases) the only releases which are official and supported releases are those designated as non-prerelease and specifically mentioned to be official releases.
+
+**WARNING**: Do not use `master` for production, instead pick a supported release version.
+
+## Creating a Release
+
+Creating a releases is as simple as cutting a tag and making a [Github release](https://help.github.com/en/github/administering-a-repository/creating-releases).
+
+If the release is meant to be an official KSphere supported release, ensure it's not marked as a pre-release, follows the `v{MAJOR}.{MINOR}.{PATCH}` [semver](https://semver.org/) pattern, and ensure the language in the title and description indicate "Official Release".
```
Comment (dkoshkin) [inline README.md]: Don't we want `v{MAJOR}.{MINOR}.{PATCH}`version to be `stable-v{MAJOR}.{MINOR}.{PATCH}-#`?

Comment (dkoshkin) [reply README.md]: cc @jieyu @jimmidyson

Comment (shaneutt) [reply README.md]: I've updated it to simply use that pattern instead of straight semver, as we don't have any other use cases right now :+1:

**Pull request merged**